collectd_plugin/collectd_plugin.py

```diff
@@ -947,5 +948,7 @@ def _should_filter(values, data, ds):
     data = {}
     collectd.register_init(init)
     collectd.register_config(config, data)
-    collectd.register_write(write, data)
+    if data.get("monitor_endpoint") or (
+            data.get("carbon_host") and data.get("carbon_port")):
+        collectd.register_write(write, data)
```

Comment:
did you test this on a ssnode?  Or do you want to package it and see what happens?

Reply:
Untested -- just trying to address your concern about continuing to pump data into a `Queue` with no `writer` thread to read from it.